### PR TITLE
Feat: Timetable upload page

### PR DESCRIPTION
**Pull request by arathibiju**
## Description
This PR resolves issue #17. This PR adds a basic Upload Page component that can be used to render a basic timetable upload page according to the designs on the figma board. The component is currently not fully functional and no styling has been applied as that will be addressed in a later PR. 

Material UI components and layouts have been used where possible to display the different buttons and text. The text field will store the typed text into a state variable inside the component. The link will redirect the user to a site where they can download their calendar file.

## How has this been tested?
The Upload Page component has basic render tests that ensure all necessary components are displayed correctly.

## Screenshots
image

## Checklist
*(Leave blank if not applicable)*

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/components/Timetable/__tests__/UploadPage.test.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import UploadPage from '../UploadPage';
+
+describe('On UploadPage render', () => {
```
Comment (scorpionknifes) [inline src/components/Timetable/__tests__/UploadPage.test.jsx:6]: Add a test for clicking submit

**Review by dlee432 — APPROVED** [review]
LGTM!

**Review by ThomasHolsterUOA — APPROVED** [review]
LGTM

File: src/components/Timetable/UploadPage.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      <TextField
+        id="outlined-input"
+        aria-label="Enter URL"
+        label="Enter your calendar URL"
+        value={calendarURL}
+        onChange={handleChange}
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <IconButton edge="end" color="primary" aria-label="submit">
```
Comment (scorpionknifes) [inline src/components/Timetable/UploadPage.jsx:38]: would be nice if there is a form for this submit or the button calls a function.

e.g sendTimetableICS(link) and this is pass through the props.

Comment (arathibiju) [reply src/components/Timetable/UploadPage.jsx:38]: Good idea, I'll pass in a function as a prop.

Comment (arathibiju) [reply src/components/Timetable/UploadPage.jsx]: @scorpionknifes Have added it now

File: src/components/Timetable/UploadPage.jsx [hunk 3]
```diff
@@ -35,7 +37,7 @@ export default function UploadPage() {
         InputProps={{
           endAdornment: (
             <InputAdornment position="end">
-              <IconButton edge="end" color="primary" aria-label="submit">
+              <IconButton edge="end" color="primary" aria-label="submit" onClick={sendTimetableURL}>
```
Comment (lin8231) [inline src/components/Timetable/UploadPage.jsx:40]:
```suggestion
              <IconButton edge="end" color="primary" aria-label="submit" onClick={() => sendTimetableURL(calendarURL)}>
```

**Review by lin8231 — CHANGES_REQUESTED** [review]

File: src/components/Timetable/UploadPage.jsx (hunk 3 above)
Comment (scorpionknifes) [reply src/components/Timetable/UploadPage.jsx:40]: @arathibiju 
would be nice if you can make it a method
```js
const handleSubmit = () => {
   sendTimetableURL(calendarURL)}
}
```

Comment (scorpionknifes) [reply src/components/Timetable/UploadPage.jsx:40]: Also would be nice to add tests for submitting

Comment (arathibiju) [reply src/components/Timetable/UploadPage.jsx]: Hiya, I've added some changes as requested as well as a sample test. Let me know if this is alright or if more changes are needed @scorpionknifes

**Review by scorpionknifes — APPROVED** [review]

**Pull request merged**